### DISCUSSION TITLE
301 redirect doesn't update feed URL

### DISCUSSION THREAD

**Discussion by kaimac1** [spacecowboy/Feeder#353] [Ideas]
(sorry if this is the wrong place to post this!)

Hello,

I see that someone is using this reader to pull my RSS feed. The location of the feed was moved with a 301 (permanent) redirect, which is followed, but the reader does not update its URL for the feed, so it is always requesting the old location. Most other readers I have seen have been able to honour the 301. I can keep the redirect in place, but if I was moving to a different domain (for example) I would lose this user. 

    [17/Aug/2024:12:35:09 +0000] "GET /blog/index.xml HTTP/1.1" 301 194 "-" "SpaceCowboys Android RSS Reader / 2.6.30(315)"
    [17/Aug/2024:12:35:09 +0000] "GET /posts/feed.xml HTTP/1.1" 200 2030 "-" "SpaceCowboys Android RSS Reader / 2.6.30(315)"
    ...some time later...
    [17/Aug/2024:15:37:25 +0000] "GET /blog/index.xml HTTP/1.1" 301 194 "-" "SpaceCowboys Android RSS Reader / 2.6.30(315)"
    [17/Aug/2024:15:37:25 +0000] "GET /posts/feed.xml HTTP/1.1" 200 2030 "-" "SpaceCowboys Android RSS Reader / 2.6.30(315)"

**Comment by spacecowboy** [comment]
Hi, thanks for reporting.

It is by design - but not by choice.

Reason can be seen here:
[                // Not changing feed url because I don't want to override auth or token params](https://github.com/spacecowboy/Feeder/blob/65fef39a3bee4fe4d88fcd5af05c1f657964f980/app/src/main/java/com/nononsenseapps/feeder/model/RssLocalSync.kt#L367)

TLDR; a long time ago there was a bug related to auto-updating the feed urls if they contained authorization or some url params. So it was disabled as an easy solution.

For most feeds it could be auto updated though, and probably should.
